plugins/woocommerce/includes/class-wc-cart-session.php

```diff
@@ -706,4 +714,81 @@ private function cart_has_shippable_products() {
 		}
 		return false;
 	}
+
+	/**
+	 * Get removed cart contents with time-based expiration (5 minutes).
+	 *
+	 * @return array Processed removed cart contents.
+	 */
+	private function get_processed_removed_cart_contents() {
```

Comment:
Note that this is unnecessary if we keep `clean_up_removed_cart_contents`, but leaving this in for comparison for now.

I'll remove this if we agree that `clean_up_removed_cart_contents` is safe enough.